ec2/spark_ec2.py

```diff
@@ -750,7 +754,7 @@ def setup_cluster(conn, master_nodes, slave_nodes, opts, deploy_ssh_key):
                'mapreduce', 'spark-standalone', 'tachyon']
 
     if opts.hadoop_major_version == "1":
-        modules = filter(lambda x: x != "mapreduce", modules)
+        modules = list(filter(lambda x: x != "mapreduce", modules))
```

Comment:
>  I saw that there was a desire to move it over to argparse?

Yes: [SPARK-6218](https://issues.apache.org/jira/browse/SPARK-6218)

> I also have a major linguistic nit to pick with the asymmetry between launch/destroy.

Feel free to bring it up on the [dev list](https://spark.apache.org/community.html) or in a JIRA.